reVX/turbine_flicker/turbine_flicker.py

```diff
@@ -0,0 +1,563 @@
+            Shifts along axis 1 from building location to pixels to be excluded
+        """
+        # ensure shadow_flicker array is regularly shaped
+        shadow_flicker = cls._check_shadow_flicker_arr(shadow_flicker)
+
+        # normalize by number of time-steps to match shadow flicker results
+        flicker_threshold /= 8760
+        shape = shadow_flicker.shape
+        row_shifts, col_shifts = np.where(shadow_flicker > flicker_threshold)
+        check = (np.any(np.isin(row_shifts, [0, shape[0] - 1]))
+                 or np.any(np.isin(col_shifts, [0, shape[1] - 1])))
+        if check:
+            msg = ("Turbine flicker exceeding {} appears to extend beyond the "
+                   "FlickerModel domain! Please increase the "
+                   "FLICKER_ARRAY_LEN and try again!")
+            logger.error(msg)
+            raise RuntimeError(msg)
+
+        row_shifts -= shape[0] // 2
+        col_shifts -= shape[1] // 2
+
+        return row_shifts, col_shifts
+
+    @staticmethod
+    def _get_building_indices(excl_fpath, building_layer, gid,
+                              resolution=128, building_threshold=0):
+        """
+        Find buildings in sc point sub-array and convert indices to full
+        exclusion indices
+
+        Parameters
+        ----------
+        excl_fpath : str
+            Filepath to exclusions h5 file. File must contain "building_layer"
+            and "tm_dset".
+        building_layer : str
+            Exclusion layer containing buildings from which turbine flicker
+            exclusions will be computed.
+        gid : int
+            sc point gid to extract buildings for
+        resolution : int, optional
+            SC resolution, must be input in combination with gid,
+            by default 128
+        building_threshold : float, optional
+            Threshold for exclusion layer values to identify pixels with
+            buildings, values are % of pixel containing a building,
+            by default 0
+
+        Returns
+        -------
+        row_idx : ndarray
+            Axis 0 indices of building in sc point sub-array in full exclusion
+            array
+        col_idx : ndarray
+            Axis 1 indices of building in sc point sub-array in full exclusion
+            array
+        shape : tuple
+            Exclusion shape
+        """
+        with ExclusionLayers(excl_fpath) as f:
+            shape = f.shape
+            row_slice, col_slice = MeanWindDirectionsPoint.get_agg_slices(
+                gid, shape, resolution)
+
+            sc_blds = f[building_layer, row_slice, col_slice]
+
+        row_idx = np.array(range(*row_slice.indices(row_slice.stop)))
+        col_idx = np.array(range(*col_slice.indices(col_slice.stop)))
+        bld_row_idx, bld_col_idx = np.where(sc_blds > building_threshold)
+
+        return row_idx[bld_row_idx], col_idx[bld_col_idx], shape
+
+    @classmethod
+    def _exclude_turbine_flicker(cls, gid, excl_fpath, res_fpath,
```

Comment:
Does this method apply the shadow-based exclusion to _all possible_ exclusion pixels using the buildings in the specified SC point gid (including excluding pixels outside of the specified SC point gid)? Or does it only look at buildings within and resulting exclusions within the specified SC point?

Reply:
All of these methods are of a single SC point gid. It will only map based on buildings within the SC Point gid sub-section of the exclusions array, BUT it will exclude pixels outside of the sub-section, if for example a building is right no the edge.

Reply:
Okay fantastic. That's exactly what I was getting at. Great. Maybe if you feel like it, add that note to the docstring that it will exclude pixels outside the sub-array